doc/tutorials/fermat_rule_reg.rst

```diff
@@ -0,0 +1,84 @@
+.. _fermat_rule_reg:
+
+======================================================
+Mathematics Behind L1 Regularization and Fermat's Rule
+======================================================
+
+This tutorial presents the mathematics behind solving the optimization problem
+:math:`\min f(x) + \lambda \|x\|_1` and demonstrates why the solution is zero when
+:math:`\lambda` is greater than the infinity norm of the gradient of :math:`f` at zero, therefore justifying the choice in skglm of
+
+.. code-block::
```

Comment:
Add a sentence saying that parametrizing the régularisation as a fraction of this, eg 0.01 alpha max, is more convenienti